src/resizing.jl

```diff
@@ -18,7 +18,10 @@ function restrict{T,N}(A::AbstractArray{T,N}, dim::Integer)
         return A
     end
     newsz = ntuple(i->i==dim?restrict_size(size(A,dim)):size(A,i), Val{N})
-    out = Array{typeof(A[1]/4+A[2]/2),N}(newsz)
+    # FIXME: The following line fails for interpolations because
```

Comment:
Interesting. I haven't coded with Interpolations recently, but if it supports what we need then
```
R = CartesianRange(indices(A))
f = A[first(R)]
```
should work. Though perhaps the better strategy would to add support for `first` to Interpolations.